apps/src/lab2/levelEditors/aichatSettings/ModelSelectionFields.tsx

```diff
@@ -0,0 +1,117 @@
+        )
+      )
+    );
+  const selectedModelId = aichatSettings.initialCustomizations.selectedModelId;
+
+  return (
+    <FieldSection
+      fieldName="selectedModelId"
+      labelText="Model Selection"
+      description="The initial model selected for the chatbot."
+      inputType="custom"
+      inputNode={
+        <>
+          <SimpleDropdown
+            labelText=""
+            onChange={e => {
+              setModelSelectionValues(
+                unionModelIds(additionalAvailableModelIds, e.target.value),
+                e.target.value
+              );
+            }}
+            items={modelDescriptions.map(model => {
+              return {value: model.id, text: model.name};
+            })}
+            selectedValue={selectedModelId}
+            name="model"
+            size="s"
+          />
+          <br />
+          <BodyFourText>
+            <i>
+              Models available to the student to select from and compare to each
+              other. If making this setting editable, it's best practice to
+              select at least 2 models so students have something to compare -
+              otherwise, consider making this read only or hidden.
+            </i>
+          </BodyFourText>
+          {modelDescriptions.map(model => {
+            return (
+              <div key={model.id} className={moduleStyles.fieldRow}>
+                <Checkbox
+                  key={model.id}
+                  name={model.id}
+                  label={model.name}
+                  checked={
+                    additionalAvailableModelIds.has(model.id) ||
+                    model.id === selectedModelId
+                  }
+                  disabled={
+                    model.id === selectedModelId ||
+                    shouldDisableAdditionalModelSelection
+                  }
+                  onChange={e => {
+                    if (e.target.checked) {
+                      setAdditionalAvailableModelIds(
+                        additionalAvailableModelIds.add(e.target.name)
+                      );
+                      setModelSelectionValues(
+                        unionModelIds(
+                          additionalAvailableModelIds,
+                          selectedModelId
+                        ),
+                        selectedModelId
+                      );
+                    } else {
+                      additionalAvailableModelIds.delete(e.target.name);
+                      setAdditionalAvailableModelIds(
+                        additionalAvailableModelIds
+                      );
+                      setModelSelectionValues(
+                        unionModelIds(
+                          additionalAvailableModelIds,
+                          selectedModelId
+                        ),
+                        selectedModelId
+                      );
+                    }
+                  }}
```

Comment:
ditto as above - not memoized

Reply:
+1, I think would help with readability too to move this elsewhere (and memoize)

Reply:
Pulled the onChange functions and items calculation up above the return, and I think I have the dependencies and stuff correct. LMK if it looks OK!